app/code/Bogdank/SupportChat/Observer/ChatUserLogin.php

```diff
@@ -20,29 +21,50 @@ class ChatUserLogin implements ObserverInterface
      */
     private $messageCollectionFactory;
 
+    /**
+     * @var \Magento\Framework\DB\TransactionFactory $transactionFactory
+     */
+    private $transactionFactory;
+
     /**
      * ChatUserLogin constructor.
-     * @param GenerateHash $generateHash
+     * @param ChatHashManager $generateHash
      * @param \Bogdank\SupportChat\Model\ResourceModel\SupportMessage\CollectionFactory $messageCollectionFactory
+     * @param \Magento\Framework\DB\TransactionFactory $transactionFactory
      */
     public function __construct(
-        \Bogdank\SupportChat\Model\GenerateHash $generateHash,
-        \Bogdank\SupportChat\Model\ResourceModel\SupportMessage\CollectionFactory $messageCollectionFactory
-    ) {
+        \Bogdank\SupportChat\Model\ChatHashManager $generateHash,
+        \Bogdank\SupportChat\Model\ResourceModel\SupportMessage\CollectionFactory $messageCollectionFactory,
+        \Magento\Framework\DB\TransactionFactory $transactionFactory
+    )
+    {
         $this->generateHash = $generateHash;
         $this->messageCollectionFactory = $messageCollectionFactory;
+        $this->transactionFactory = $transactionFactory;
     }
+
     public function execute(Observer $observer)
     {
         /** @var \Magento\Customer\Model\Data\Customer $customer */
         $customer = $observer->getData('customer');
-        $this->updateChatMessages($customer);
+        $this->updateChatMessagesData($customer);
```

Comment:
@holmatro90 Exception must not prevent customer from logging in, but any issues with updating messages must be saved to the log file via `Psr\Log\LoggerInterface`